docs/adrs/multus.md

```diff
@@ -0,0 +1,36 @@
+# Record architecture decisions
+
+Date: 2024-02-07
+
+## Status
+
+Discussing
+
+## Context
+
+### Multus
+
+Multus is a CNI multiplexer that allows pods to have multiple network interfaces. We have users that are operating K3s + Multus but it is not super obvious how to configure it to work with K3s and how to add the additional pieces needed (e.g. IPAM or additional CNI plugins). We could facilitate this by creating an integration with Multus. Moreover, Multus deployments normally come with a CNI IPAM plugin to distribute IPs to the extra interfaces. The most popular one among Multus users is Whereabouts, so we will take the decision to use that one for this integration.
+
+### Design suggestion
+
+The suggestion is to include a new `--multus` boolean flag. When that flag gets activated, we deploy a Kubernetes job that will run in that node and will:
+* Run the [cni-plugins image](https://github.com/rancher/image-build-cni-plugins) which will deploy in the node all the widely used CNI binaries (e.g. macvlan), the multus binary and the whereabouts binary
+* Install the required multus CRD `network-attachment-definitions.k8s.cni.cncf.io` and the whereabouts CRDs `ippools.whereabouts.cni.cncf.io` & `overlappingrangeipreservations.whereabouts.cni.cncf.io`
```

Comment:
I would probably do like we do with disabled components in RKE2, and ship these as a flat manifest, and then just add them to the `--disable` list if `--multus` isn't set.